onboard/nav/avoidance.hpp

```diff
@@ -0,0 +1,54 @@
+#ifndef AVOIDANCE_HPP
+#define AVOIDANCE_HPP
+
+#include "rover.hpp"
+
+class StateMachine;
+
+class Avoidance {
+public:
+    /*************************************************************************/
+    /* Public Member Functions */
+    /*************************************************************************/
+    Avoidance(StateMachine* stateMachine_)
+    : stateMachine(stateMachine_),
+    mJustDetectedObstacle(false) {}
+
+    virtual ~Avoidance() {}
+
+    void updateObstacleAngle( double bearing );
+
+    NavState run( Rover * phoebe, const rapidjson::Document& mRoverConfig );
+
+    virtual Odometry createAvoidancePoint( Rover * phoebe, const double distance ) = 0;
+
+    virtual NavState executeTurnAroundObs( Rover* phoebe, const rapidjson::Document& mRoverConfig ) = 0;
+
+    virtual NavState executeDriveAroundObs( Rover* phoebe ) = 0;
+
+
+
+protected:  // TODO
+    /*************************************************************************/
+    /* Protected Member Variables */
+    /*************************************************************************/
+    StateMachine* stateMachine;
```

Comment:
I changed the name to roverStateMachine because it's not really a member variable. I don't want it to get confused with the other pointer. (Also, I think this is what you asked above when changing from stateMachine( stateMachine_ ) to roverStateMachine( stateMachine_ ) in the constructor.